db-developments/python/aggregate.py

```diff
@@ -0,0 +1,105 @@
+from jinja2 import Template
+from datetime import datetime
+from pathlib import Path
+import os
+import sys
+from sqlalchemy import text
+from utils import engine
+
+geoms = {
+    "block": {
+        "source_column": "bctcb{{decade}}",
+        "output_column": "bctcb{{decade}}",
+        "additional_columns": [
+            ("geoid", "cenblock{{decade[:-2]}}")
+        ],
+        "group_by": ["boro", "bctcb{{decade}}", "cenblock{{decade}}"],
+        "join_table": "dcp_cb{{decade}}"
+    },
+    "cdta": {
+        "source_column": "cdta{{decade}}",
+        "output_column": "cdta{{decade}}",
+        "additional_columns": [
+            ("cdtaname", "cdtaname{{decade[:-2]}}")
+        ],
+        "join_table": "dcp_cdta{{decade}}"
+    },
+    "commntydst": {
+        "source_column": "comunitydist",
+        "join_table": "dcp_cdboundaries",
+        "right_join_column": "borocd::TEXT"
+    },
+    "councildst": {
+        "source_column": "councildist",
+        "join_table": "dcp_councildistricts",
+        "right_join_column": "coundist::TEXT"
+    },
+    "nta": {
+        "source_column": "nta{{decade}}",
+        "output_column": "nta{{decade}}",
+        "additional_columns": [
+            ("ntaname", "ntaname{{decade[:-2]}}")
+        ],
+        "join_table": "dcp_nta{{decade}}"
+    },
+    "tract": {
+        "source_column": "bct{{decade}}",
+        "output_column": "bct{{decade}}",
+        "additional_columns": [
+            ("geoid", "centract{{decade}}")
+        ],
+        "group_by": ["boro", "bct{{decade}}", "centract{{decade}}"],
+        "join_table": "dcp_ct{{decade}}",
+        "right_join_column": "boroct2020"
+    }
+}
+
+if __name__ == "__main__":
+    filename = sys.argv[1]
+    decade = sys.argv[2]  # e.g. "10" or "20"
+    CAPTURE_DATE = os.environ.get("CAPTURE_DATE")
+
+    # Get current year
+    current_year = datetime.today().year
+
+    # SQL Template
+    with open(filename, 'r') as f:
+        sql = f.read()
+
+    try:
+        geom=sys.argv[3]
+    except:
```

Comment:
a little risky not specifying the "expected" exception. would it be `except IndexError:`?

Reply:
Good point. I'll check and implement